src/DotNetOutdated/V3NuGetClient.cs

```diff
@@ -1,61 +1,92 @@
 using System;
 using System.Collections.Generic;
-using System.Threading.Tasks;
-using Newtonsoft.Json.Linq;
-using NuGet.Versioning;
 using System.Linq;
+using System.Net.Http;
+using System.Threading;
+using System.Threading.Tasks;
+using NuGet.Common;
+using NuGet.Configuration;
+using NuGet.Protocol;
+using NuGet.Protocol.Core.Types;
 
 namespace DotNetOutdated
 {
-    public abstract class V3NuGetClient
+    public class V3NugetClient 
     {
-        protected abstract Task<JObject> GetResource(string name);
+        private readonly HttpClient httpClient;
+
+        public V3NugetClient(HttpClient httpClient)
+        {
+            this.httpClient = httpClient ?? throw new ArgumentNullException(nameof(httpClient));
+        }
 
-        public async Task<PackageInfo> GetPackageInfo(string packageName)
+        public async Task<PackageInfo> GetPackageInfo(string name)
         {
-            var json = await this.GetResource($"{packageName.ToLower()}/index.json");
+            var logger = new Logger();
+            var providers = new List<Lazy<INuGetResourceProvider>>();
+            providers.AddRange(Repository.Provider.GetCoreV3());
+            var packageSource = new PackageSource("https://api.nuget.org/v3/index.json");
+            var sourceRepository = new SourceRepository(packageSource, providers);
 
-            if (json == null)
+            var packageMetadataResource = await sourceRepository.GetResourceAsync<PackageMetadataResource>();
+            IEnumerable<IPackageSearchMetadata> searchMetadata = await packageMetadataResource.GetMetadataAsync(name, true, true, new SourceCacheContext(), logger, CancellationToken.None);
+
+            if(!searchMetadata.Any())
             {
                 return null;
             }
-            var versions = new List<SemanticVersion>();
 
-            var items = json["items"].AsJEnumerable();
-            if (items[0]["items"] != null)
+            var versions = searchMetadata.Select(package => package.Identity.Version).ToList();
```

Comment:
This isn't quite correct as your search query may match multiple packages (example: `json` will match both  `newtonsoft.json` and `json`). Also some NuGet feeds like Sleet return ALL packages regardless of the search query. I would recommend querying [the package metadata resource](https://docs.microsoft.com/en-us/nuget/api/registration-base-url-resource) instead.